src/core/annotation.js

```diff
@@ -138,6 +138,10 @@ class AnnotationFactory {
       attachments,
       xfaDatasets,
       collectFields,
+      needAppearances:
+        !collectFields &&
+        ((acroForm instanceof Dict && acroForm.get("NeedAppearances")) ||
+          false),
```

Comment:
While there's still a bunch of regressions that we need to address, blocking this PR, I couldn't help to look at these lines and think about the overall readability.

How about we make the following changes, to improve both this and the old code?
```diff
diff --git a/src/core/annotation.js b/src/core/annotation.js
index 748ae7f2b..b7b52e219 100644
--- a/src/core/annotation.js
+++ b/src/core/annotation.js
@@ -119,6 +119,7 @@ class AnnotationFactory {
       return undefined;
     }

+    const acroFormDict = acroForm instanceof Dict ? acroForm : Dict.empty;
     const id =
       ref instanceof Ref ? ref.toString() : `annot_${idFactory.createObjId()}`;

@@ -134,14 +135,12 @@ class AnnotationFactory {
       subtype,
       id,
       pdfManager,
-      acroForm: acroForm instanceof Dict ? acroForm : Dict.empty,
+      acroForm: acroFormDict,
       attachments,
       xfaDatasets,
       collectFields,
       needAppearances:
-        !collectFields &&
-        ((acroForm instanceof Dict && acroForm.get("NeedAppearances")) ||
-          false),
+        !collectFields && acroFormDict.get("NeedAppearances") === true,
       pageIndex,
     };
```